src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1022,6 +1022,10 @@ public void Copy()
         /// </summary>
         public void Paste()
         {
+            //clear the selection so we can put the
+            //paste contents in
+            DynamoSelection.Instance.ClearSelection();
+
```

Comment:
I think this state change should be front & center, rather than buried in local variable definitions.